qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -57,3 +61,50 @@ def post(self):
         else:
             raise HTTPError(400, 'Unknown sample template action: %s' % action)
         self.write(result)
+
+
+def build_sample_summary(study_id, user_id, visible=None):
```

Comment:
Also, i'm unsure what did you changed in this function, but this is still asking for all the prep templates info - i.e. you are still transferring all the information. If you want, we can meet later today to find a way to improve this function.